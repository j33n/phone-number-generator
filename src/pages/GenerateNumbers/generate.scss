.generator-button {
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 70px auto;
    border-radius: 25px;
    background: #249292;
    color: #fff;

    width: 40%;

    &:hover {
        background: #007172;
        cursor: pointer;
    }
}

.loader {
    height: 20px;
    background: red;
    border-radius: 10px;
    width: 50%;
}

.alert {
    font-size: 12px;
    color: #fff;

    &--success {
        color: #4bb543;
    }

    &--danger {
        color: #bb2124
    }

    &--warning {
        color: #ffcc00;
    }
}