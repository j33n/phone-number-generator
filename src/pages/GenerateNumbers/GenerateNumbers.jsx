import React, { PureComponent } from 'react';

import NumberTable from '../../components/NumberTable/NumberTable';

import './generate.scss';
import ScrollEnd from '../../components/ScrollEnd/ScrollEnd';

class GenerateNumbers extends PureComponent {
  state = {
    numbersGenerated: 0,
  }

  formZeroedNumber = (number, numberOfZeroes) => {
    let zeroes;
    let combinedZeroes = '';
    for (zeroes = 1; zeroes <= numberOfZeroes; zeroes += 1) {
      combinedZeroes = `${combinedZeroes}0`;
    }
    return combinedZeroes + number;
  }

  stringifyNumber = (number, length) => {
    const strNumber = number.toString(10);
    if (strNumber.length < length) {
      return this.formZeroedNumber(number, length - strNumber.length);
    }
    return number.toString(10);
  }

  triggerGenerator = () => {
    const arrayLength = new Uint32Array(10000);
    const randomNumbers = window.crypto.getRandomValues(arrayLength);

    const generatedNumber = [];
    randomNumbers.forEach((randomNumber) => {
      generatedNumber.push(this.stringifyNumber(randomNumber, 10));
    });
    if (generatedNumber.length === 10000) {
      // Save numbers generated to localStorage
      localStorage.setItem('numbers', generatedNumber);
      this.setState({
        numbersGenerated: generatedNumber.length,
        allNumbers: generatedNumber,
      });
    }
  };

  render() {
    const { numbersGenerated, allNumbers } = this.state;
    return (
      <div className="container">
        <ScrollEnd size={32} />
        <div className="generator-button" onClick={this.triggerGenerator} type="button" role="presentation">Generate Random Phone Numbers</div>
        {!!numbersGenerated && (
          <div className="alert-container">
            <span className="alert alert--success">
              {`${numbersGenerated} Random Phone Numbers were Generated`}
            </span>
          </div>
        )}
        {!!numbersGenerated && allNumbers.length === 10000 && (
          <NumberTable numbers={allNumbers} />
        )}
      </div>
    );
  }
}

export default GenerateNumbers;
