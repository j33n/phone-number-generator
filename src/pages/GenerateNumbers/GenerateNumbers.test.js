import React from 'react';
import {
  shallow
} from 'enzyme';
import renderer from 'react-test-renderer';

import GenerateNumbers, {
  formZeroedNumber
} from './GenerateNumbers';

describe('GenerateNumbers', () => {
  const component = <GenerateNumbers />;

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should form zeroed numbers', () => {
    expect(formZeroedNumber('123', '3')).toEqual('000123');
  });

  //   it('should update height based on clientHeight', () => {
  //     const shallowComponent = shallow(component);

//     shallowComponent.setState({
//       height: 0
//     });
//     // shallowComponent.instance().dimensionRef = {
//     //   current: {
//     //     clientHeight: 10,
//     //   }
//     // };
//     expect(tree.state('height')).toEqual(10);
//   });
});
