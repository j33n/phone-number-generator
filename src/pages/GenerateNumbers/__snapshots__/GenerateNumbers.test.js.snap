// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateNumbers renders without crashing 1`] = `<GenerateNumbers />`;
